raster/src/main/scala/geotrellis/raster/render/ColorBreaks.scala

```diff
@@ -21,30 +21,73 @@ import java.util.Locale
 
 /**
   * ColorBreaks describes a way to render a raster into a colored image.
+  */
+trait ColorBreaks extends Serializable {
+  def colors: Array[Int]
+  def length: Int
+  def replaceColors(newColors: Array[Int]): ColorBreaks
+  def mapColors(f: Int => Int): ColorBreaks
+  def toColorMap(): ColorMap = toColorMap(ColorMapOptions.Default)
+  def toColorMap(options: ColorMapOptions): ColorMap
+}
+
+/**
+  * IntColorBreaks describes a way to render a raster into a colored image.
   *
-  * This class defines a set of value ranges and assigns a color to
+  * This class defines a set of Int value ranges and assigns a color to
   * each value range.
   *
   * @param limits  An array with the maximum value of each range
   * @param colors  An array with the color assigned to each range
   */
-case class ColorBreaks(limits: Array[Int], colors: Array[Int]) {
+class IntColorBreaks(val limits: Array[Int], val colors: Array[Int]) extends ColorBreaks {
   assert(limits.length == colors.length)
   assert(colors.length > 0)
 
-  val lastColor = colors(colors.length - 1)
-
   def length = limits.length
 
-  def get(z: Int): Int = {
-    var i = 0
-    val last = colors.length - 1
-    while (i < last) {
-      if (z <= limits(i)) return colors(i)
-      i += 1
-    }
-    lastColor
+  def replaceColors(newColors: Array[Int]): ColorBreaks =
+    new IntColorBreaks(limits, newColors)
+
+  def mapColors(f: Int => Int): ColorBreaks =
+    new IntColorBreaks(limits, colors.map(f))
+
+  def toColorMap(options: ColorMapOptions = ColorMapOptions.Default): ColorMap =
+    ColorMap(limits, colors, options)
+
+  override def toString = {
+    val limitsStr = limits.mkString("Array(", ", ", ")")
+    val colorsStr =
+      colors
+        .map("%08x" formatLocal(Locale.ENGLISH, _))
+        .mkString("Array(", ", ", ")")
+    s"IntColorBreaks($limitsStr, $colorsStr)"
   }
+}
+
+/**
+  * DoubleColorBreaks describes a way to render a raster into a colored image.
+  *
+  * This class defines a set of Double value ranges and assigns a color to
+  * each value range.
+  *
+  * @param limits  An array with the maximum value of each range
+  * @param colors  An array with the color assigned to each range
+  */
+class DoubleColorBreaks(val limits: Array[Double], val colors: Array[Int]) extends ColorBreaks {
```

Comment:
Good call on serializable. but not sure it's worth making it a case class (adds bytecode for equals, creates an object, not necessary if we're not going to use it)